iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp

```diff
@@ -263,14 +271,129 @@ struct ECInstanceMethodSymbolsProvider : IECSymbolProvider
+            return ExpressionStatus::WrongNumberOfArguments;
+            }
+
+        for (size_t i = 0; i < 3; ++i)
+            {
+            if (!args[i].IsECValue() || !args[i].GetECValue()->IsString())
+                {
+                DIAGNOSTICS_LOG(DiagnosticsCategory::ECExpressions, LOG_TRACE, LOG_ERROR, Utf8PrintfString("Invalid type of argument %" PRIu64 " - expected a string", (uint64_t)i));
+                return ExpressionStatus::WrongType;
+                }
+            }
+
+        Context ctx = *static_cast<Context*>(context);
+
+        Utf8String queryFormat;
+        ExpressionStatus stat = GetRelatedDisplayLabelQuery(queryFormat, args[0].GetECValue()->GetUtf8CP(), args[1].GetECValue()->GetUtf8CP(), args[2].GetECValue()->GetUtf8CP());
+        if (ExpressionStatus::Success != stat)
+            return stat;
+
+        for (IECInstancePtr const& instance : instanceData)
+            {
+            Utf8PrintfString query(queryFormat.c_str(), instance->GetClass().GetECSqlName().c_str());
+
+            ECSqlStatement stmt;
+            ECSqlStatus status = stmt.Prepare(ctx.m_connection.GetECDb().Schemas(), ctx.m_connection.GetDb(), query.c_str());
+            if (!status.IsSuccess())
+                DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::ECExpressions, "Failed to prepare related display label query");
+
+            ECInstanceId id;
+            ECInstanceId::FromString(id, instance->GetInstanceId().c_str());
+            status = stmt.BindId(1, id);
+            if (!status.IsSuccess())
+                DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::ECExpressions, "Failed to bind related display label query parameter");
+
+            if (BE_SQLITE_ROW == QueryExecutorHelper::Step(stmt))
+                {
+                evalResult.InitECValue() = ECValue(stmt.GetValueText(0));
+                ECEXPRESSIONS_EVALUATE_LOG(LOG_TRACE, Utf8PrintfString("ECInstanceMethodSymbolsProvider::GetRelatedDisplayLabel: Result: %s", evalResult.ToString().c_str()).c_str());
+                return ExpressionStatus::Success;
+                }
+            }
+
+        evalResult.InitECValue().SetToNull();
+        ECEXPRESSIONS_EVALUATE_LOG(LOG_TRACE, Utf8PrintfString("ECInstanceMethodSymbolsProvider::GetRelatedDisplayLabel: Result: %s", evalResult.ToString().c_str()).c_str());
+        return ExpressionStatus::Success;
+        }
+        
 protected:
     Utf8CP _GetName() const override {return "ECInstanceMethods";}
     void _PublishSymbols(SymbolExpressionContextR context, bvector<Utf8String> const& requestedSymbolSets) const override
         {
         context.AddSymbol(*MethodSymbol::Create("IsOfClass", NULL, &IsOfClass));
+        void* methodContext = const_cast<Context*>(&m_context);
+        context.AddSymbol(*MethodSymbol::Create("GetRelatedDisplayLabel", nullptr, &GetRelatedDisplayLabel, methodContext));

```

Comment:
Please add this function to [this table](https://github.com/iTwin/itwinjs-core/blob/efcafe30035c2469c3cace676e5ffcfd1e19ed5b/docs/presentation/advanced/ECExpressions.md?plain=1#L272) in associated `itwinjs-core` pr.

Reply:
https://github.com/iTwin/itwinjs-core/pull/7124

Reply:
please make sure to complete the `itwinjs-core` PR after this one